docker/Dockerfile

```diff
@@ -61,5 +52,14 @@ RUN --mount=type=bind,source=.git,target=${PUDL_REPO}/.git \
     # Run the PUDL setup script so we know where to read and write data
     ${CONDA_RUN} pudl_setup
 
+# Install awscli2
+RUN ${CONDA_RUN} curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" && \
```

Comment:
Isn't this is still going to try and run the system `curl` and presumably fail because we're no longer installing `curl`? If we want to use `curl` from `conda` I think we'll need to add `curl` to the list of stuff that's installed by `mamba` (currently we install `libcurl` because it's an indirect dependency)

Reply:
I added `curl` to the `[tool.conda-lock.dependencies]` portion of `pyproject.toml`. When I rebuild the env locally its still using my system `curl`. Is that the right place too add curl in `pyproject.toml`?

Reply:
Testing it locally now.